src/Compiler/TaskManager.cpp

```diff
@@ -228,10 +221,17 @@ void TaskManager::reset() {
   }
 }
 
-void TaskManager::rollBack(Task *t) {
-  const auto &roll_back = m_rollBack.value(t);
-  for (const auto &task : roll_back) {
-    task->setStatus(TaskStatus::None);
+void TaskManager::cleanDownStreamStatus(Task *t) {
+  for (auto it{m_taskQueue.begin()}; it != m_taskQueue.end(); ++it) {
+    if (*it == t) {
+      // In case clean action, clean parent is required.
```

Comment:
What's the behavior when we don't have clean action for a task ?

Reply:
nothing happened in this case